api/src/repositories/spatial-repository.ts

```diff
@@ -488,34 +488,11 @@ export class SpatialRepository extends BaseRepository {
         this._whereBoundaryIntersects(criteria.boundary, 'geography', qb1);
       })
       .with('with_user_security_transform_exceptions', (qb6) => {
-        // Build an array of the users spatial security transform exceptions
-        qb6
-          .select(knex.raw('array_agg(suse.security_transform_id) as user_security_transform_exceptions'))
-          .from('system_user_security_exception as suse')
-          .where('suse.system_user_id', this.connection.systemUserId());
+        this._buildSpatialSecurityExceptions(knex, qb6, this.connection.systemUserId());
       })
       .select(
         // Select either the non-secure or secure spatial component from the search results, based on whether or not the record had security transforms applied to it and whether or not the user has the necessary exceptions
-        knex.raw(
-          `
-            jsonb_build_object(
-              'submission_spatial_component_id',
-                wfscwst.submission_spatial_component_id,
-              'spatial_data',
-                -- when: the user's security transform ids array contains all of the rows security transform ids (user has all necessary exceptions)
-                -- then: return the spatial component
-                -- else: return the secure spatial component if it is not null (secure, insufficient exceptions), otherwise return the spatial compnent (non-secure, no exceptions required)
-                case
-                  when
-                    wuste.user_security_transform_exceptions @> wfscwst.spatial_component_security_transforms
-                  then
-                    wfscwst.spatial_component
-                  else
-                    coalesce(wfscwst.secured_spatial_component, wfscwst.spatial_component)
-                end
-            ) spatial_component
-          `
-        )
+        knex.fromRaw(this._buildSelectForSecureNonSecureSpatialComponents(knex))
```

Comment:
I wonder if we need to pass `knex` through here, or if we can just get a new instance inside the function and it will still play nice?

We aren't leveraging any of the Knex database connection stuff, its only ever being used for its query building capabilities, so I would hope that using separate instances of knex would be fine. Maybe?